push_config.go

```diff
@@ -0,0 +1,183 @@
+// Copyright 2019 Aporeto Inc.
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//     http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package elemental
+
+import (
+	"fmt"
+	"net/url"
+)
+
+// A PushFilter represents an abstract filter for filtering out push notifications. This is now aliased to PushConfig as a
+// result of re-naming the type.
+//
+// Deprecated: use the new name PushConfig instead
+type PushFilter = PushConfig
+
+// A PushConfig represents an abstract filter for filtering out push notifications.
+//
+// The 'IdentityFilters' field is a mapping between a filtered identity and the string representation of an elemental.Filter.
+// A client will supply this attribute if they want fine-grained filtering on the set of identities that they are filtering on.
+// If this attribute has been supplied, the identities passed to 'IdentityFilters' must be a subset of the identities passed to
+// 'Identities'; passing in identities that are not provided in the 'Identities' field will be ignored.
+type PushConfig struct {
+	Identities      map[string][]EventType `msgpack:"identities" json:"identities"`
+	IdentityFilters map[string]string      `msgpack:"filters"    json:"filters"`
```

Comment:
⚠️  **This is an important naming decision** ⚠️ 

What do we think about the name here for the new attribute? I just called it 'filters'.

Any suggestions? This is basically final as changing it in the future will be a major major breaking change.

@t00f @primalmotion

Reply:
filters is good. that's how it's called everywhere